src/modules/iotjs_module_tcp.c

```diff
@@ -400,6 +401,12 @@ JS_FUNCTION(Listen) {
   int err = uv_listen((uv_stream_t*)(iotjs_tcpwrap_tcp_handle(tcp_wrap)),
                       backlog, OnConnection);
 
+  if (err < 0) {
+    jerry_value_t jerror = iotjs_create_uv_exception(err, "listen");
+    jerry_value_set_error_flag(&jerror);
+    return jerror;
+  }
```

Comment:
See https://github.com/Rokid/ShadowNode/blob/00a011f48b22ee3f0efae8460f01313a6db1f97f/src/js/net.js#L483 
removing the changes and using `process._createUVException()`

Reply:
As the error shall be considered as a uv call exception, I think it's better to create the js error object right after the uv calls.
Maybe the native module could return the error object rather than throwing it?